### PR TITLE
#1387 fix img for systemjs and refactoring apiservice

### PR DESCRIPTION
**Pull request by magemello**
**Please check if the PR fulfills these requirements**
```
[ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
[ ] Tests for the changes have been added (for bug fixes / features)
[ ] Docs have been added / updated (for bug fixes / features)
```

**What kind of change does this PR introduce?** (check one with "x")
```
[ x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] Documentation
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

#1387 , I have also refactored the authService and apiService... the injection pattern was used in wrong way.

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ ] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: ng2-components/ng2-alfresco-core/src/services/AlfrescoAuthentication.service.ts [hunk 1]
```diff
@@ -20,16 +20,14 @@ import { LogService } from './log.service';
 import { SettingsService } from './settings.service';
 import { StorageService } from './storage.service';
 import { AlfrescoApiService } from './alfresco-api.service';
-import { AuthService } from './auth.service';
 
 /** @deprecated AlfrescoAuthenticationService is deprecated. Use AuthService instead */
 @Injectable()
-export class AlfrescoAuthenticationService extends AuthService {
+export class AlfrescoAuthenticationService {
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-core/src/services/AlfrescoAuthentication.service.ts]: The idea was that both old and new service still function, and old one is raising a working. If we wrap AuthService into the AlfrescoAuthenticationService (and emulate apis) then both services will work the same.

Comment (magemello) [inline ng2-components/ng2-alfresco-core/src/services/AlfrescoAuthentication.service.ts]: If we keep it in this way it's going to be easy have double instance of the same class, with inspected behaviour, one for AlfrescoAuthenticationService and another for AuthService.

Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-core/src/services/AlfrescoAuthentication.service.ts]: Yes, having "extends" is wrong, I was thinking over injecting AuthService into AlfrescoAuthenticationService, i.e.

```ts
@Injectable()
export class AlfrescoAuthenticationService {
  constructor(private authService: AuthService) {}

  login(username: string, password: string) {
     return this.authService.login(username, password);
  }
}
```

Comment (magemello) [inline ng2-components/ng2-alfresco-core/src/services/AlfrescoAuthentication.service.ts]: I think you will still get error at build time if you were using the old type...

**Pull request merged**